frontend/src/pages/ExecutionList.tsx

```diff
@@ -132,26 +138,30 @@ class ExecutionList extends Page<{}, ExecutionListState> {
       if (!this.executionTypesMap || !this.executionTypesMap.size) {
         this.executionTypesMap = await this.getExecutionTypes();
       }
-      const executions = await this.getExecutions(listOperationOpts);
+      const executions = this.props.isGroupView
+        ? await this.getExecutions()
+        : await this.getExecutions(listOperationOpts);
       this.clearBanner();
       const flattenedRows = this.getFlattenedRowsFromExecutions(request, executions);
+      const grouppedRows = this.getGroupedRowsFromExecutions(request, executions);
       // TODO(jlyaoyuli): Consider to support grouped rows with pagination.
       this.setState({
         executions,
-        rows: flattenedRows,
+        expandedRows: this.props.isGroupView ? grouppedRows.expandedRows : new Map(),
+        rows: this.props.isGroupView ? grouppedRows.collapsedRows : flattenedRows,
       });
     } catch (err) {
       this.showPageError(serviceErrorToString(err));
     }
     return listOperationOpts.getNextPageToken();
   }
 
-  private async getExecutions(listOperationOpts: ListOperationOptions): Promise<Execution[]> {
+  private async getExecutions(listOperationOpts?: ListOperationOptions): Promise<Execution[]> {
     try {
       const response = await this.api.metadataStoreService.getExecutions(
         new GetExecutionsRequest().setOptions(listOperationOpts),
       );
-      listOperationOpts.setNextPageToken(response.getNextPageToken());
+      listOperationOpts?.setNextPageToken(response.getNextPageToken());
```

Comment:
Can we bring back the test suite in `ExecutionList.test.tsx` to with parameter `isGroupView={isGroupView}` to make sure ExecutionList is compatible with both default and grouped views?